paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -149,22 +152,22 @@ internal class PaymentSheetPlaygroundViewModel(
 
             snapshot.saveToSharedPreferences(getApplication())
 
-            state.value = PlaygroundState.Customer(
-                snapshot = snapshot,
-                adapter = CustomerAdapter.create(
-                    context = getApplication(),
-                    customerEphemeralKeyProvider = {
-                        fetchEphemeralKey(snapshot)
-                    },
-                    setupIntentClientSecretProvider = if (
-                        snapshot[CustomerSheetPaymentMethodModeDefinition] == PaymentMethodMode.SetupIntent
-                    ) {
-                        { customerId -> createSetupIntentClientSecret(snapshot, customerId) }
-                    } else {
-                        null
-                    }
-                )
+            val adapter = CustomerAdapter.create(
+                context = getApplication(),
+                customerEphemeralKeyProvider = {
+                    fetchEphemeralKey(snapshot)
+                },
+                setupIntentClientSecretProvider = if (
+                    snapshot[CustomerSheetPaymentMethodModeDefinition] == PaymentMethodMode.SetupIntent
+                ) {
+                    { customerId -> createSetupIntentClientSecret(snapshot, customerId) }
+                } else {
+                    null
+                }
             )
+
+            state.value = PlaygroundState.Customer(snapshot = snapshot)
+            customerAdapter.value = adapter
```

Comment:
Storing the `adapter` as part of the `PlaygroundState` caused the `ViewModel` to leak. Moved it into its own flow.

Reply:
That's weird. Was the view model holding onto it, or what?

Reply:
The methods passed into the `CustomerAdapter` reference `ViewModel` attributes (for example, `settings` in order to the update them). 

This was a one way to solve. We could instead change the implementation to prefetch `ephemeralKey` & `setupIntentClientSecret` when preparing `CustomerSheet`.